components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java

```diff
@@ -272,5 +334,145 @@ protected User getUser(Map eventProperties, UserStoreManager userStoreManager){
         return user;
     }
 
+    private void preSetUserClaimsOnEmailUpdate(Map<String, String> claims, UserStoreManager userStoreManager, User user)
+            throws IdentityEventException {
+
+        Utils.clearEmailVerifyTemporaryClaim();
+
+        if (MapUtils.isEmpty(claims)) {
+            // Not required to handle in this handler.
```

Comment:
Shall we add a debug log here? WDYT?

Reply:
I don't think adding a debug log would add any value here.